### ISSUE TITLE
Astromenace 1.3.2 rendering problems

### ISSUE THREAD

**Issue by petevine** [lunixbochs/glshim#133]
The visually stuning game, made in Russia, runs beautifully on an old x86 using OpenGL 1.4 (nvidia binary drivers).
But on Odroid C1 (x11, Mali) the game seems to be working, but all's black except for the splash screen which happens to be completely white.
I've copied the settings from that x86 box but no luck: 

````
./AstroMenace
AstroMenace 1.3.2 130712

Open XML file: /home/odroid/.config/astromenace/amconfig.xml
VFS file was opened ./gamedata.vfs

Font initialized: DATA/FONT/LinBiolinumBold.ttf

Vendor     : OpenAL Community
Renderer   : OpenAL Soft
Version    : 1.1 ALSOFT 1.14
ALut ver   : 1.1

Current Video Mode: 1280x1024 16bit 

Xinerama/TwinView detected.
Screen count: 1
Screen #0: (0, 0) x (1280, 1024)

libGL:loaded: libGLESv1_CM.so
libGL:loaded: libEGL.so
libGL: built on Aug 30 2015 11:08:44
SDL version: 1.2.15
Set video mode: 1280 x 1024 x 16

Vendor     : ARM
Renderer   : Mali-450 MP
Version    : 1.4 glshim wrapper
libGL: GL_INVALID_ENUM when calling glGet<GL_INT>(GL_MAJOR_VERSION)
libGL: GL_INVALID_ENUM when calling glGet<GL_INT>(GL_MINOR_VERSION)
OpenGL Version    : 0.0

Max texture height: 4096 
Max texture width: 4096 
Max lights: 8 
Max multitexture supported: 8 textures.
Shaders unsupported.
````

Would the failed OpenGL version check be of any importance?

**Comment by lunixbochs** [comment]
The easiest solution here would be me trying the game myself. I'll take a look.

**Comment by petevine** [comment]
I've just tried it with a different version of glshim (ssvb probably) and it identifies OpenGL as 1.0 and is able to render some elements making the game barely playable.

**Comment by lunixbochs** [comment]
Are you on the unstable branch? Try both master and unstable.

**Comment by petevine** [comment]
Yeah, unstable from a week ago.

**Comment by petevine** [comment]
Is it ok to just shuffle the builds without recompiling the game binaries?

**Comment by lunixbochs** [comment]
yep

**Comment by petevine** [comment]
Glest crashes with :
````
./glest: symbol lookup error: ./glest: undefined symbol: glDrawRangeElements
````
after switching to master or that older version

**Comment by lunixbochs** [comment]
that's fine, this issue is about Astromenace

**Comment by petevine** [comment]
I know, I meant switching libs can cause problems. Was it because of libGLU?

**Comment by lunixbochs** [comment]
no. master doesn't have glDrawRangeElements implemented. I just want to know if Astromence works fine on master.

**Comment by petevine** [comment]
All right, I answered that already - sorry for my imprudent editing :)

**Comment by lunixbochs** [comment]
you should also try ptitseb's glshim fork

**Comment by petevine** [comment]
I can see ptitseb is also building preload which means this could be the one I was referring to earlier.
However I'm unable to build it from the latest master:

/tmp/glshim-master/src/glx/streaming.c:4:20: fatal error: bc_cat.h: No such file or directory
 #include <bc_cat.h>
                    ^
What am I missing?

**Comment by lunixbochs** [comment]
his fork is different enough from mine I can't really provide support on it, it's just it at a different spot so has different compatibility

**Comment by petevine** [comment]
Certainly. Does the include look familiar to you? I can't seem to find a download anywhere.

**Comment by petevine** [comment]
Certainly. It seems the file was provided inside the snapshot after all. 

Both games seem to work fine with ptitseb's fork!

**Comment by lunixbochs** [comment]
The point of these issues is to track upstream glshim support, as ptitseb's fork is very different.

**Comment by petevine** [comment]
In that case I could give you a crash outside of glshim (caused by it) present in both repos. Interested?

**Comment by lunixbochs** [comment]
Yep, I'm interested in any relevant bug reports.

(Haven't been working on glshim as much in the past couple of months because I've been very busy with Usercorn, but glshim should be back in the rotation very soon as the Pyra is fast approaching.)
